drivers/gpio/gpio_intel.c

```diff
@@ -114,17 +115,64 @@ struct gpio_intel_data {
 	uint32_t pad_base;
 
 	sys_slist_t cb;
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(acpi_hid)
+	uint32_t num_pins;
+	uint32_t pad_owner_reg;
+	uint32_t host_owner_reg;
+	uint32_t intr_stat_reg;
+	uint32_t base_num;
+#endif
 };
 
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(acpi_hid)
+#define GPIO_REG_BASE_GET(dev) DEVICE_MMIO_NAMED_GET(dev, reg_base)
+
+#define REG_GPI_INT_STS_BASE_GET(data) data->intr_stat_reg
+
+#define REG_GPI_INT_EN_BASE_GET(data) (data->intr_stat_reg + 0x20)
+
+#define PIN_OFFSET_GET(dev) (0)
+
+#define GPIO_PAD_OWNERSHIP_GET(data, pin, offset) data->pad_owner_reg + ((pin / 8) * 0x4)
+
+#define REG_PAD_HOST_SW_OWNER_GET(data) data->host_owner_reg
+
+#define GPIO_BASE_GET(cdf) (0)
+
+#define GPIO_INTERRUPT_BASE_GET(cfg) (0)
+
+#define GPIO_GET_PIN_MAX(dev) ((struct gpio_intel_data *)dev->data)->num_pins
```

Comment:
Just as a general principle, it's good to always enclose in parenthesis the usage of macro parameters. Otherwise you might get surprises and hard-to-debug bugs if someone calls the macro with a parameter which isn't atomically evaluated.